Example/Example Swift/ComponentAPICoordinator.swift

```diff
@@ -392,23 +413,30 @@ extension ComponentAPICoordinator: UINavigationControllerDelegate {
         if fromVC is ReviewViewController && operation == .pop {
             reviewScreen = nil
             if let document = pages.first?.document {
-                documentService?.delete(document)
+                documentService?.remove(document: document)
             }
         }
         
-        if fromVC is AnalysisViewController && operation == .pop {
+        if fromVC is AnalysisViewController {
             analysisScreen = nil
-            documentService?.cancelAnalysis()
+            if operation == .pop {
+                documentService?.cancelAnalysis()
+            }
         }
         
-        if toVC is CameraViewController && (fromVC is ReviewViewController || fromVC is AnalysisViewController) {
+        if toVC is CameraViewController &&
+            (fromVC is ReviewViewController ||
+             fromVC is AnalysisViewController ||
+             fromVC is ImageAnalysisNoResultsViewController) {
```

Comment:
Hmm, so all this condition does is check if the app is in "Single page" mode (as opposed to multi page) and the user is cancelling the processing?

If so, it is a little fragile. Instead of knowing that it's "Single page", you check for all possible view controller configurations where that's the case. There's a danger that if you change something in the screen flow, this will no longer be the right check.

Reply:
It is not only Single page but also when a PDF or a QRCode is analyzed in Multi page mode.
For me the `navigationController(_ navigationController:operation:fromVC:toVC:)` method is the key in the navigation since every transition can be intercepted, so this must be done here and not in the `backTo` or `goTo` actions of every screen. It is still a little bit fragile but less than the other cases, and makes sense to do it in this method.

Reply:
I'm not questioning whether or not the code should be here, but the way the condition is checked. It's fragile to look into the combination of view controllers and how they transition from one to another.

It would have been better for the Coordinator to already know if it's the single or multipage mode (or the type of document) and knew what to do without deriving the current situation purely based to the screen transitions.

Reply:
That's the first think I thought, but still need to know what is the current screen, which could lead to some spaghetti code (when every screen is shown/removed)